patches/minecraft/net/minecraft/client/gui/recipebook/RecipeBookGui.java.patch

```diff
@@ -1,15 +1,118 @@
 --- a/net/minecraft/client/gui/recipebook/RecipeBookGui.java
 +++ b/net/minecraft/client/gui/recipebook/RecipeBookGui.java
-@@ -88,7 +88,7 @@
+@@ -52,6 +52,9 @@
+    protected final RecipeItemHelper field_193965_u = new RecipeItemHelper();
+    private int field_193966_v;
+    private boolean field_199738_u;
++   private net.minecraftforge.client.UpDownButton up;
++   private net.minecraftforge.client.UpDownButton down;
++   private int offset;
+ 
+    public void func_201520_a(int p_201520_1_, int p_201520_2_, Minecraft p_201520_3_, boolean p_201520_4_, RecipeBookContainer<?> p_201520_5_) {
+       this.field_191888_F = p_201520_3_;
+@@ -88,13 +91,13 @@
        this.func_205702_a();
        this.field_193018_j.clear();
  
 -      for(RecipeBookCategories recipebookcategories : ClientRecipeBook.func_216769_b(this.field_201522_g)) {
-+      for(RecipeBookCategories recipebookcategories : this.field_201522_g.getRecipeBookCategories()) {
++      for(net.minecraftforge.common.extensions.IForgeRecipeBookCategory<?> recipebookcategories : this.field_201522_g.getRecipeBookCategories()) {
           this.field_193018_j.add(new RecipeTabToggleWidget(recipebookcategories));
        }
  
-@@ -417,7 +417,7 @@
+       if (this.field_191913_x != null) {
+          this.field_191913_x = this.field_193018_j.stream().filter((p_209505_1_) -> {
+-            return p_209505_1_.func_201503_d().equals(this.field_191913_x.func_201503_d());
++            return p_209505_1_.getRecipeBookCategory().equals(this.field_191913_x.getRecipeBookCategory());
+          }).findFirst().orElse((RecipeTabToggleWidget)null);
+       }
+ 
+@@ -104,7 +107,18 @@
+ 
+       this.field_191913_x.func_191753_b(true);
+       this.func_193003_g(false);
++      // remove tabs that does not contains recipes
```

Comment:
Patches should be without comments.
In my eyes the patch is clear in its intention, when you read the change.
As such it is a higher priority to keep the patches as small as possible and then comments are not really needed and just bloat the patch.

Could you remove it?

Reply:
Removed comments